qemu/tests/qemu_guest_agent.py

```diff
@@ -126,19 +125,23 @@ def _check_ga_pkg(self, session, cmd_check_pkg):
                               logging.info)
         s, o = session.cmd_status_output(cmd_check_pkg)
         if s == 0 and self.params.get("os_variant", "") == 'rhel8':
-            # qemu-guest-agent-2.12.0-88.module+el8.1.0+4233+bc44be3f.x86_64
-            error_context.context("Check if the installed pkg is the latest"
+            error_context.context("Check if the installed pkg is the specific"
                                   " one for rhel8 guest.", logging.info)
             version_list = []
-            build_latest = re.sub(r'/', '-', self.qga_pkg_latest_url)
-            for pkg in [o, build_latest]:
+            full_qga_rpm_info = session.cmd("readlink %s" % self.qga_pkg_path)
+            build_specific = re.sub(r'/', '-', full_qga_rpm_info)
+            for pkg in [o, build_specific]:
                 pattern = r"guest-agent-(\d+.\d+.\d+-\d+).module"
                 qga_v = re.findall(pattern, pkg, re.I)[0]
                 version_list.append(qga_v)
-            logging.info("The installed and the latest pkg version is"
+            logging.info("The installed and the specific pkg version is"
                          " %s", version_list)
-            if version_list[1] != version_list[0]:
-                return False
+            if version_list[1]:
+                if version_list[1] != version_list[0]:
+                    return False
+            else:
+                self.test.error("It is recommended that you install"
+                                "a specific version of qga rpm in advance")
```

Comment:
@vivianQizhu , thanks. Code has been updated.

Reply:
I have a bit different opinion about this.

If can not get version info with the pattern, i prefer to use '', instead of append nothing.   Other wise, it looks puzzled for the following code if there len is 1.
```
logging.info("The installed and the specific pkg "
                   "version is %s", version_list)
```

By the way, it is not absolute that `len(version_list) < 2` is caused by no proper qga be specified, since no installed qga can turn to this result too.

So i suggest use `` when failed to get the version with pattern, and compare version_list[1] and  version_list[0].

Reply:
If no installed qga `s, o = session.cmd_status_output(cmd_check_pkg)` `s == 0` will be `False` then it will not step into here.
And if you assume there is exception that it might not be installed, then it is possible that they are both `""` and then the function will return `True`, is that expected?
However, both okay for me if the logic is correct and expected and also the code is not in low efficiency.

Reply:
If no installed qga `s, o = session.cmd_status_output(cmd_check_pkg)` `s == 0` will be `False` then it will not step into here.
And if you assume there is exception that it might not be installed, then it is possible that they are both `""` and then the function will return `True`, is that expected?
However, both okay for me if the logic is correct and expected and also the code is not in low efficiency.

Reply:
> If no installed qga `s, o = session.cmd_status_output(cmd_check_pkg)` `s == 0` will be `False` then it will not step into here.

Yes, you are right, in this case it will return False directly, this is the expected result. I miss this before.

> And if you assume there is exception that it might not be installed, then it is possible that they are both `""` and then the function will return `True`, is that expected?

Not the expected, and the explanation above can avoid this, i just do not want to see the debug info like the following example when no proper qga package be provided.

`The installed and the specific pkg version is [$INSTALLED_PKG]`

> However, both okay for me if the logic is correct and expected and also the code is not in low efficiency.

Reply:
Eh~  Get the installed qga pkg is not equal to can get install qga version, so we still can not avoid both `""`.
Bases on this, i am ok with this logic ~

Reply:
@yanan-fu If move the `logging` line above to `qga_v = re.findall(pattern, pkg, re.I)`, anyway when it steps there both the installed one and `full_qga_rpm_info` are all exist, simply log it like `"Found qga pkg: %s" % pkg`, what do you think?

Reply:
@vivianQizhu In this way, it can not distinguish the installed pkg and specified pkg from the log. No friendly for debugging, do you think this is a problem ? Eh~   not a big issue :)

Or we can use dict instead of list to save the version info ?

Reply:
@yanan-fu I have thought about dict too, but yes not a big deal, so let's leave it to @6-dehan at his discretion.